primitives/runtime/src/lib.rs

```diff
@@ -257,6 +262,72 @@ pub fn storage_value_key(pallet_prefix: &str, value_name: &str) -> StorageKey {
 	StorageKey(final_key)
 }
 
+#[derive(Encode, Decode, TypeInfo, PalletError)]
+pub enum OwnedBridgeModuleError {
+	/// All pallet operations are halted.
+	Halted,
+}
+
+pub trait OwnedBridgeModule<T: frame_system::Config> {
+	const NAME: &'static str;
+	const OPERATING_MODE_KEY: &'static str;
```

Comment:
I admit that without this key it'll be too confusing to read "Setting operating mode to false" instead of "Setting operating mode (IsHalted = false)." for the GRANDPA module :) WDYT about introducing something like `enum <GoodName> { Normal, Halted }` here and using it for both GRANDPA and parachains finality pallets? then we don't need this associated const at all.

I know that isn't a part of the issue, but I can recall at least two issues with improper use of `IsHalted`. And right now it is a good time to make this change, because we have no active deployments (apart from Rialto/Millau), so no runtime upgrades required. If you also think it is better, then let's probably file a new issue for that and impl in a separate PR? It must be marked with `breaksruntime` label, so that we (and other teams) won't miss it.

Reply:
Sounds good to me. Having an `enum` instead of `bool` would make things more clear. I oepend an issue for this: https://github.com/paritytech/parity-bridges-common/issues/1479